godtools/Views/TractElements/BaseTractElement.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  BaseTractElement.swift
+//  godtools
+//
+//  Created by Ryan Carlson on 4/26/17.
+//  Copyright © 2017 Cru. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SWXMLHash
+
+class BaseTractElement: UIView {
+    struct Standards {
+        static let xMargin = CGFloat(8.0)
+        static let yMargin = CGFloat(8.0)
+        static let xPadding = CGFloat(0.0)
+        static let yPadding = CGFloat(0.0)
+        
+        static let screenWidth = UIScreen.main.bounds.size.width
+        static let textContentWidth = UIScreen.main.bounds.size.width - xMargin * CGFloat(2)
+    }
+    
+    weak var parent: BaseTractElement?
+    var elements:[BaseTractElement]?
+    var yStartPosition: CGFloat = 0.0
+    var maxHeight: CGFloat = 0.0
+    private var _height: CGFloat = 0.0
+    var height: CGFloat {
+        get {
+            return _height
+        }
+        set {
+            _height = newValue
+        }
+    }
+    var width: CGFloat {
+        if (self.parent != nil) {
+            return (self.parent?.width)!
```

Comment:
If you force unwrap parent, then you don't have to unwrap the whole statement.
````swift
return self.parent!.width
````